components/proxy/src/test/java/com/hotels/styx/metrics/reporting/graphite/GraphiteReporterServiceTest.java

```diff
@@ -17,48 +17,98 @@
 
     @BeforeEach
     public void setUp() {
         meterRegistry = new CompositeMeterRegistry();
         log = new LoggingTestSupport(GraphiteReporterService.class);
-        service = new GraphiteReporterService.Builder()
-                .meterRegistry(meterRegistry)
-                .serviceName("Graphite-Reporter-test")
-                .prefix("test")
-                .host("localhost")
-                .port(8080)
-                .reportingIntervalMillis(10)
-                .build();
     }
 
     @AfterEach
     public void stop() {
         log.stop();
     }
 
+    GraphiteReporterService.Builder serviceBuilder() {
+        return new GraphiteReporterService.Builder()
+                .meterRegistry(meterRegistry)
+                .serviceName("Graphite-Reporter-test")
+                .host("localhost")
+                .port(8080)
+                .reportingIntervalMillis(10);
+    }
+
     @Test
     public void logsWhenServiceStarts() {
+        GraphiteReporterService service = serviceBuilder().build();
+
         try {
             StyxFutures.await(service.start());
             assertThat(log.lastMessage(), is(loggingEvent(INFO, "Graphite service started, service name=\"Graphite\\-Reporter\\-test\"")));
         } finally {
             StyxFutures.await(service.stop());
         }
     }
+
+    @Test
+    public void supportsGraphiteWithTags() {
+        GraphiteReporterService service = serviceBuilder().tagsEnabled(true).build();
+
+        try {
+            StyxFutures.await(service.start());
+            GraphiteMeterRegistry graphiteRegistry = (GraphiteMeterRegistry) meterRegistry.getRegistries().iterator().next();
+            assertThat(graphiteRegistry.config().namingConvention(), instanceOf(GraphiteDimensionalNamingConvention.class));
+        } finally {
+            StyxFutures.await(service.stop());
+        }
+
+    }
+
+    @Test
+    public void supportsGraphiteWithoutTags() {
+        GraphiteReporterService service = serviceBuilder().tagsEnabled(false).build();
+
+        try {
+            StyxFutures.await(service.start());
+            GraphiteMeterRegistry graphiteRegistry = (GraphiteMeterRegistry) meterRegistry.getRegistries().iterator().next();
+            assertThat(graphiteRegistry.config().namingConvention(), instanceOf(GraphiteHierarchicalNamingConvention.class));
+        } finally {
+            StyxFutures.await(service.stop());
+        }
+
+    }
+
+    @Test
+    public void defaultsToNoTagSupport() {
+        GraphiteReporterService service = serviceBuilder().build();
+
+        try {
+            StyxFutures.await(service.start());
+            GraphiteMeterRegistry graphiteRegistry = (GraphiteMeterRegistry) meterRegistry.getRegistries().iterator().next();
+            assertThat(graphiteRegistry.config().namingConvention(), instanceOf(GraphiteHierarchicalNamingConvention.class));
+        } finally {
+            StyxFutures.await(service.stop());
+        }
+
```

Comment:
Could we lose the extra whitespace here and the multiple blank lines at the end of the class, please?